sdks/python/apache_beam/io/iobase.py

```diff
@@ -972,14 +979,15 @@ def expand(self, pcoll):
                        'or be a PTransform. Received : %r' % self.sink)
 
 
-class WriteImpl(ptransform.PTransform):
+class WriteImpl(ptransform.PTransform[InT, None]):
```

Comment:
Or should this be a PTransform[pvalue.PCollection[InT], None]?

Reply:
IMHO, that would be redundant.  It won't change the behavior of the type checks, and it would add verbosity to _a lot_ of class definitions, since there are numerous `PTransforms` throughout the code.

Reply:
oh, also, it will require an import of the `apache_beam.pvalue` module into all of those modules as well.  you won't be able to protect it inside a `typing.TYPE_CHECKING` block since it's used in the class definition.

Reply:
But some transforms take and return other things than PCollections. Maybe if T is shorthand for PCollection[T] but that seems ambiguous.